### PR TITLE
Book detail page

### LINKED ISSUE #166: Update book details page

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes #166 and [WWW-100](https://jira.nypl.org/browse/WWW-100).

Updates the book details page and fixes the issue with getting all the filters from the data model.

Example links:
* http://staff-picks-qa.us-east-1.elasticbeanstalk.com/books-music-movies/recommendations/best-books/staff-picks/2017-09-01/0062438891-the-legend-of-rock-paper-scissors
* http://staff-picks-qa.us-east-1.elasticbeanstalk.com/books-music-movies/recommendations/best-books/staff-picks/2018-03-01/9780765395726-a-red-peace
* http://staff-picks-qa.us-east-1.elasticbeanstalk.com/books-music-movies/recommendations/best-books/staff-picks/2018-01-01/9781885008572-the-crane-girl-based-on-japanese-folktales

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
Gah, I didn't commit them. Will update.

**Comment by EdwinGuzman** [comment]
Updated and added tests.

File: test/unit/Book.test.js [hunk 1]
```diff
@@ -85,10 +86,20 @@ describe('Book Component', () => {
     });
 
     it('should render the pick description <p> element with text', () => {
-      const text = component.find('.book-item-description');
-      expect(text.length).to.equal(1);
-      expect(text.find('p').length).to.equal(1);
-      expect(text.find('p').text()).to.equal(pickObject.reviews[0].text);
+      const description = component.find('.book-item-description');
+      expect(description.length).to.equal(1);
+      expect(description.length).to.equal(1);
```
Comment (gkallenberg) [inline test/unit/Book.test.js]: double check here?

File: test/unit/Book.test.js [hunk 2]
```diff
@@ -85,10 +86,20 @@ describe('Book Component', () => {
...
+      expect(description.text()).to.equal(pickObject.reviews[0].text);
+    });
+
+    it('should render the pick reviewer', () => {
+      const reviewer = component.find('.book-item-picked-by');
+      const reviewerText = `Staff Pick By: ${pickObject.reviews[0].reviewerName}, ` +
+        `${pickObject.reviews[0].reviewerLocation.prefLabel}`;
+
+      expect(reviewer.length).to.equal(1);
+      expect(reviewer.length).to.equal(1);
```
Comment (gkallenberg) [inline test/unit/Book.test.js]: double check here?

**Comment by EdwinGuzman** [comment]
@gkallenberg yes, removed. Thanks 👀 .

**Comment by gkallenberg** [comment]
One other thing I found which may or may not be a design problem, but at what appears to be the 483px breakpoint, the image min-height is set to 470px which stretches the display when wrapping the "picked by" text. Can that min-height be changed to be more responsive or is that a best practice. I don't mind whitespace, but it seems a bit odd for that particular display break.

**Comment by ricardoom** [comment]
@gkallenberg, Could you hook up a screen shot?

**Comment by gkallenberg** [comment]
@ricardogalvez 
screen gap

**Comment by ricardoom** [comment]
cool one.

**Comment by holingpoon** [comment]
Travis is in green anyway, I would just like to know if I should be concerned about the warnings returned by the test suite: [BookPage Component](https://gist.github.com/holingpoon/ef7ccb86fe2a1812ef5a1bf0c632c527)

**Comment by EdwinGuzman** [comment]
Updated the Action warning, the other is not as bad so leaving it there.

**Pull request merged**